jolie/src/main/java/jolie/runtime/expression/InlineTreeExpression.java

```diff
@@ -57,7 +58,17 @@ public Operation cloneOperation( TransformationReason reason )
 		@Override
 		public void run( Value inlineValue )
 		{
-			path.getValue( inlineValue ).deepCopyWithLinks( expression.evaluate() );
+			if ( expression instanceof VariablePath ) {
+				Object myObj = (((VariablePath) expression).getValueOrValueVector());
+				if ( myObj instanceof Value ) {
```

Comment:
please make this an if ( myObj instanceof Value ) { ... } else if ( myObj instanceof ValueVector ) { ... } else { throw new RuntimeException( "incomplete case analysis" ); }